src/main/java/lembas/core/LembasCore.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (C) 2019 Lembas Modding Team
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package lembas.core;
+
+import cpw.mods.fml.common.*;
+import cpw.mods.fml.common.Mod.*;
+import cpw.mods.fml.common.event.*;
+import lembas.core.api.client.LembasCoreClient;
+import lembas.core.api.server.LembasCoreServer;
+import lembas.core.internal.client.LembasCoreClientImpl;
+import lembas.core.internal.common.LembasCoreCommonImpl;
+import lembas.core.internal.server.LembasCoreServerImpl;
+
+/**
+ * The main mod file of Lembas Core. Everything which can be accessed from the
+ * outside (modid, version, getClient(), ...) is considered to be a part of the
+ * API. Internal implementation details (for example the proxy implementation
+ * classes) ARE NOT PART OF THE API.
+ * 
+ * @author CraftedMods
+ *
+ */
+@Mod(modid = LembasCore.MODID, version = LembasCore.VERSION, name = "Lembas Core", acceptedMinecraftVersions = "*")
+public class LembasCore
+{
+
+    public static final String MODID = "lembas-core";
+    public static final String VERSION = "@version@"; // Injected by gradle
+
+    @Instance(value = LembasCore.MODID)
+    public static LembasCore instance;
+
+    @SidedProxy(clientSide = "lembas.core.internal.client.LembasCoreClientImpl", serverSide = "lembas.core.internal.server.LembasCoreServerImplF", modId = LembasCore.MODID)
+    public static LembasCoreCommonImpl proxy;
+
+    private LembasCoreClientImpl client;

```

Comment:
Really nitpicking here, but it is better to keep the core API part as distant from the implementation as possible (1)
```suggestion
    private LembasCoreClient client;
```

Reply:
This is actually necessary (and this the javadoc comments explicitely state that the private things are NOT part of the API), because the callbacks for the events are not part of the API, because there's no point for outside mods to call those, so this is necessary

Reply:
So that actually means there can only be one possible implementation of the API?

Reply:
LembasCore is a file which contains both API and implementation - LembasCoreClientImpl is considered implementation, where LembasCoreClient in the getClient() function is considered API.

Reply:
(The file isn't part of the API or INTERNAL packages because of that)

Reply:
Ah, okay.